soweego/linker/blocking.py

```diff
@@ -75,22 +70,23 @@ def full_text_query_block(goal: str, catalog: str, wikidata_series: pd.Series, c
     return samples_index
 
 
-def _run_full_text_query(terms: list, target_entity: constants.DB_ENTITY, boolean_mode=False, limit=5):
-    tids = set()
-    if isinstance(terms, str):
-        terms = [terms]
-    LOGGER.debug("Full-text query terms: %s", terms)
+def _multiprocessing_series_iterator(wikidata_series: pd.Series, target_entity: constants.DB_ENTITY) -> Iterable[
+    Tuple[str, str, constants.DB_ENTITY]]:
+    for qids, terms in wikidata_series.items():
+        yield qids, terms, target_entity
 
-    for result in tokens_fulltext_search(target_entity, boolean_mode, terms):
-        tids.add(result.catalog_id)
-        LOGGER.debug('%s result: %s',
-                     target_entity.__name__, result)
 
-    if not tids:
-        LOGGER.info('No target candidates for source query: %s', terms)
-        return nan
+def fulltext_search(qid_terms_target: Tuple[str, list, constants.DB_ENTITY]) -> Iterable[Tuple[str, str]]:
+    qid = qid_terms_target[0]
```

Comment:
```suggestion
    qid, terms, target_entity = qid_terms_target
```
Just a little bit more Pythonic 😄